pytential/symbolic/execution.py

```diff
@@ -37,6 +37,7 @@
 from pytential.symbolic.compiler import Code, Statement, Assign, ComputePotential
 
 from pytential import sym
+from pytential.symbolic.dof_desc import UNNAMED_SOURCE, UNNAMED_TARGET
 
 import logging
```

Comment:
- Delete `domains` arg on `build_matrix`
- Delete `domains` arg on `scipy_op`.

Reply:
Why? Weren't those supposed to support multiple geometries with different properties? (which, yeah, nobody uses or tests at all.. but still!)

Reply:
Hmmm, upon closer reflection, I think I may have jumped the gun on removing `domains`. I think it has a role to play still. What tripped me up is that there should be `source_domains` and `target_domains`, technically. Where it says `DEFAULT_TARGET` must be a key, instead say something like "target" must be the result of an unnamed binding, such as might result from passing a tuple to `bind`.

IOW, @alexfikl, you're completely right.